job/server/src/main/java/alluxio/job/migrate/MigrateDefinition.java

```diff
@@ -104,30 +102,15 @@
 public final class MigrateDefinition
     extends AbstractVoidJobDefinition<MigrateConfig, ArrayList<MigrateCommand>> {
   private static final Logger LOG = LoggerFactory.getLogger(MigrateDefinition.class);
-  private final FileSystem mFileSystem;
-  private final FileSystemContext mFsContext;
   private final Random mRandom = new Random();
 
   /**
    * Constructs a new {@link MigrateDefinition}.
    */
   public MigrateDefinition() {
-    mFsContext = FileSystemContext.create(ServerConfiguration.global());
-    mFileSystem = BaseFileSystem.create(mFsContext);
   }
 
-  /**
-   * Constructs a new {@link MigrateDefinition} with FileSystem context and instance.
-   *
-   * @param fsContext the {@link FileSystemContext} used by the {@link FileSystem}
-   * @param fileSystem the {@link FileSystem} client
-   */
-  public MigrateDefinition(FileSystemContext fsContext, FileSystem fileSystem) {
-    mFsContext = fsContext;
-    mFileSystem = fileSystem;
-  }
-
-  private void checkMigrateValid(MigrateConfig config) throws Exception {
+  private void checkMigrateValid(MigrateConfig config, FileSystem fs) throws Exception {
```

Comment:
Should we always pass in `Job*Context`?

Reply:
because this is a private method I only passed the necessary objects